starfish/core/image/_filter/reduce.py

```diff
@@ -0,0 +1,123 @@
+import warnings
+from copy import deepcopy
+from typing import Callable, Iterable, Optional, Union
+
+import numpy as np
+
+from starfish.core.imagestack.imagestack import ImageStack
+from starfish.core.types import Axes
+from starfish.core.util import click
+from ._base import FilterAlgorithmBase
+
+
+class Reduce(FilterAlgorithmBase):
+    """
```

Comment:
Did we decide to put the docs in the class and not the constructor?  I can't remember. :(

Reply:
This is how it's done in other filter components, but I can move it if the plan is to change it moving forward.

Reply:
IMO that is a mistake, and we should rectify it. :)

Reply:
(deleted earlier comment) I misread it.

Cool, if this is what is done elsewhere, that's fine.